pubsub/server.go

```diff
@@ -0,0 +1,75 @@
+package pubsub
+
+import (
+	"context"
+	"crypto/rand"
+	"encoding/binary"
+	"errors"
+	"sync"
+
+	"github.com/NethermindEth/juno/utils"
+	"github.com/ethereum/go-ethereum/event"
+	"github.com/sourcegraph/conc"
+)
+
+var ErrNotFound = errors.New("subscription not found")
+
+type Server struct {
+	log utils.SimpleLogger
+	wg  conc.WaitGroup
+
+	mu            sync.Mutex // protects subscriptions
+	subscriptions map[uint64]event.Subscription
+}
```

Comment:
Can you elaborate on how you imagine this would fit with the rest of the design? This looks like a registry instead of a server to me.

Reply:
Registry is a better name.

I imagine the `rpc.Handler` having a registry. It's handlers create subscriptions and call `h.registry.Subscribe` / `h.registry.Unsubscribe`. The handlers return the responses to initial subscription request (containing the id) and unsubscribe request.

Reply:
Why do we need a Registry?

Do you intend to emit geth events directly from `blockchain`?

Reply:
The registry keeps track of subscriptions so we can release them when we receive an unsubscribe request.

Reply:
> Do you intend to emit geth events directly from blockchain?

We can do that, or have callbacks. The registry should be agnostic to how we manage events.